src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/NamingStyle.cs

```diff
@@ -63,6 +63,32 @@ internal partial struct NamingStyle
                 newName, newPrefix, newSuffix, newWordSeparator, newCapitalizationScheme);
         }
 
+        public override bool Equals(object obj)
+        {
+            return obj is NamingStyle other
+                && Equals(other);
+        }
+
+        public bool Equals(NamingStyle other)
+        {
+            return ID == other.ID
```

Comment:
..

what's the point of a guid here :)   if we have a guid, isn't htat enough for equality ;-)

(note: i'm find with keeping the pr like this.  but if we have naming styles with the same guids and different values, i'm going to be a sad panda).

Reply:
Not sure why, but this matches what we did before and the perf issue is gone.